### PR TITLE
implemented hierarchy checks for kick , ban ,role commands

### PR DESCRIPTION
**Pull request by sarthak-py**
and , removed the default value `None` for member arguments , i mean let the error handler deal with it *shrugs*

### CODE REVIEW

File: src/cogs/mod.py [hunk 1]
```diff
@@ -160,7 +163,13 @@ async def unmute(self, ctx, member: disnake.Member, *, reason):
...
     @commands.bot_has_permissions(manage_roles=True)
     async def role(self, ctx, member: disnake.Member, role: disnake.Role, *, reason="No Reason Provided."):
-        await member.add_roles(role,reason=reason)
+        """ Add a role to the user"""
+        if ctx.author.top_role.position <= role.position:
+            return await ctx.send('You cannot add that role to someone!')
+        try : 
+            await member.add_roles(role,reason=reason)
+        except :
+            return await ctx.send('Unable to add that role to '+member.name)
```
Comment (Robin5605) [inline src/cogs/mod.py:172]: F-strings to maintain consistence

File: src/cogs/mod.py [hunk 2]
```diff
@@ -108,6 +109,8 @@ async def unban(
     @commands.bot_has_permissions(manage_channels=True)
     async def slowmode(self, ctx: Context, channel: disnake.TextChannel = None, slowmode: int = None):
         """Change/disable slowmode in a channel"""
+        if channel not in ctx.guild.channels:
```
Comment (Robin5605) [inline src/cogs/mod.py:112]: Redundant, the `disnake.TextChannel` wouldn't find it anyway

File: src/cogs/mod.py [hunk 3]
```diff
@@ -23,25 +22,27 @@ def __init__(self, bot: commands.Bot) -> None:
...
         self,
         ctx: Context,
-        member: disnake.Member = None,
+        member: disnake.Member ,
         reason: str = "No Reason Provided.",
     ):
         """ Kicks a member from a guild """
-        if not member:
-            return await ctx.send(
-                f"{ctx.author.mention}, please provide a member to kick."
```
Comment (Robin5605) [inline src/cogs/mod.py:32]: Reduce this to a one-liner, just reduces readability

File: src/cogs/mod.py [hunk 4]
```diff
@@ -23,25 +22,27 @@ def __init__(self, bot: commands.Bot) -> None:
...
             ctx, message="Are you sure you want to kick this user?", timeout=60
             )
 
         if choice:
-            await member.kick(reason=reason)
-            await ctx.send(f"{member.mention} has been kicked.")
+            try :
+                await member.kick(reason=reason)
+                await ctx.send(f"{member.mention} has been kicked.")
+            except :
```
Comment (Yat-o) [inline src/cogs/mod.py:44]: Bare exception clause

**Review by Robin5605 — CHANGES_REQUESTED** [review]
Please resolve the comments I have left, and await approval by another reviewer. If approved by the other reviewer, either I or the other reviewer will merge it.

File: src/cogs/mod.py (hunk 3 above)
Comment (sarthak-py) [reply src/cogs/mod.py:32]: didnt check it , added by someone else 
ill fix it though :+1:

Comment (Robin5605) [reply src/cogs/mod.py:32]: My bad, should've seen it in the red areas haha

File: src/cogs/mod.py (hunk 4 above)
Comment (sarthak-py) [reply src/cogs/mod.py]: alrighty

**Comment by Dekriel** [comment]
perhaps we can add a mention instead of name?

**Comment by sarthak-py** [comment]
> perhaps we can add a mention instead of name?

Where though?
If you mean at the "Unable to add role to ..." Part , I did it purposely because pings are annoying

**Comment by Dekriel** [comment]
> where though?
perhaps here
> return await ctx.send(f'{ctx.author} You can\'t kick **{member.name}**')

**Comment by sarthak-py** [comment]
> > where though?
> 
> perhaps here
> 
> > return await ctx.send(f'{ctx.author} You can't kick **{member.name}**')

same applies here too *shrugs*
why mention people on a command failure

**Pull request merged**